drivers/dai/intel/dmic/dmic.c

```diff
@@ -0,0 +1,673 @@
+		/* Don't stop CIC yet if one FIFO remains active */
+		if (dai_dmic_global.active_fifos_mask == 0) {
+			dai_dmic_update_bits(dmic, base[i] + CIC_CONTROL,
+					     CIC_CONTROL_SOFT_RESET_BIT |
+					     CIC_CONTROL_MIC_MUTE_BIT,
+					     CIC_CONTROL_SOFT_RESET_BIT |
+					     CIC_CONTROL_MIC_MUTE_BIT);
+		}
+		switch (dmic->dai_config_params.dai_index) {
+		case 0:
+			dai_dmic_update_bits(dmic, base[i] + FIR_CONTROL_A,
+					     FIR_CONTROL_A_MUTE_BIT,
+					     FIR_CONTROL_A_MUTE_BIT);
+			break;
+		case 1:
+			dai_dmic_update_bits(dmic, base[i] + FIR_CONTROL_B,
+					     FIR_CONTROL_B_MUTE_BIT,
+					     FIR_CONTROL_B_MUTE_BIT);
+			break;
+		}
+	}
+
+	k_spin_unlock(&dmic->lock, key);
+}
+
+const struct dai_properties *dai_dmic_get_properties(const struct device *dev,
+						     enum dai_dir dir,
+						     int stream_id)
+{
+	const struct dai_intel_dmic *dmic = (const struct dai_intel_dmic *)dev->data;
+	struct dai_properties *prop = (struct dai_properties *)dev->config;
+
+	prop->fifo_address = dmic->fifo.offset;
+	prop->dma_hs_id = dmic->fifo.handshake;
+	prop->reg_init_delay = 0;
+
+	return prop;
+}
+
+static int dai_dmic_trigger(const struct device *dev, enum dai_dir dir,
+		       enum dai_trigger_cmd cmd)
+{
+	struct dai_intel_dmic *dmic = (struct dai_intel_dmic *)dev->data;
+
+	LOG_DBG("dmic_trigger()");
+
+	if (dir != DAI_DIR_RX) {
+		LOG_ERR("dmic_trigger(): direction != DAI_DIR_RX");
+		return -EINVAL;
+	}
+
+	switch (cmd) {
+	case DAI_TRIGGER_START:
+		if (dmic->state == DAI_STATE_PAUSED ||
+		    dmic->state == DAI_STATE_PRE_RUNNING) {
+			dai_dmic_start(dmic);
+			dmic->state = DAI_STATE_RUNNING;
+		} else {
+			LOG_ERR("dmic_trigger(): state is not prepare or paused, dmic->state = %u",
+				dmic->state);
+		}
+		break;
+	case DAI_TRIGGER_STOP:
+		dai_dmic_stop(dmic, false);
+		dmic->state = DAI_STATE_PRE_RUNNING;
+		break;
+	case DAI_TRIGGER_PAUSE:
+		dai_dmic_stop(dmic, true);
+		dmic->state = DAI_STATE_PAUSED;
+		break;
+	default:
+		break;
+	}
+
+	return 0;
+}
+
+static const struct dai_config *dai_dmic_get_config(const struct device *dev, enum dai_dir dir)
+{
+	struct dai_intel_dmic *dmic = (struct dai_intel_dmic *)dev->data;
+
+	__ASSERT_NO_MSG(dir == DAI_DIR_CAPTURE);
+	return &dmic->dai_config_params;
+}
+
+static int dai_dmic_set_config(const struct device *dev,
+		const struct dai_config *cfg, const void *bespoke_cfg)
+
+{
+	struct dai_intel_dmic *dmic = (struct dai_intel_dmic *)dev->data;
+	int ret = 0;
+	int di = dmic->dai_config_params.dai_index;
+	k_spinlock_key_t key;
+
+	LOG_INF("dmic_set_config()");
+
+	if (di >= CONFIG_DAI_DMIC_HW_FIFOS) {
+		LOG_ERR("dmic_set_config(): DAI index exceeds number of FIFOs");
+		return -EINVAL;
+	}
+
+	if (!bespoke_cfg) {
+		LOG_ERR("dmic_set_config(): NULL config");
+		return -EINVAL;
+	}
+
+	__ASSERT_NO_MSG(dmic->created);
+	key = k_spin_lock(&dmic->lock);
+
+#if CONFIG_DAI_INTEL_DMIC_TPLG_PARAMS
+#error DMIC TPLG is not yet implemented
```

Comment:
@marcinszkudlinski Is the plan to merge this first with only NHLT params support and add TPLG params later, or?

Reply:
yes, there are 2 fetures missing now: TPLG and volume rampup